js/capture.js

```diff
@@ -19,13 +27,61 @@ const capture = {
       });
   },
 
+  // In the parsed whitelist JSON object (from disconnect-entitylist.json):
+  //   - 'properties' keys are first parties
+  //   - 'resources' keys are third parties
+  reformatList(whiteList) {
+    const firstPartyWhiteList = {};
+    const thirdPartyWhiteList = {};
+    let counter = 0;
+    for (const siteOwner in whiteList) {
+      const firstParties = whiteList[siteOwner].properties;
+      for (let i = 0; i < firstParties.length; i++) {
+        firstPartyWhiteList[firstParties[i]] = counter;
+      }
+      const thirdParties = whiteList[siteOwner].resources;
+      thirdPartyWhiteList[counter] = [];
+      for (let i = 0; i < thirdParties.length; i++) {
+        thirdPartyWhiteList[counter].push(thirdParties[i]);
+      }
+      counter++;
+    }
+
+    return {
+      firstPartyWhiteList,
+      thirdPartyWhiteList
+    };
+  },
+
   // Returns true if the request should be stored, otherwise false.
-  shouldStore(tab) {
+  shouldStore(tab, thirdPartyFromRequest) {
     const documentUrl = new URL(tab.url);
     // ignore about:*, moz-extension:* & non-visible tabs (like dev tools)
     if (documentUrl.protocol !== 'about:'
       && documentUrl.protocol !== 'moz-extension:'
       && tab.id !== browser.tabs.TAB_ID_NONE) {
+      // check if third party is on the whitelist (owned by the first party)
+      if (thirdPartyFromRequest) {
+        const firstPartyFromRequest = documentUrl.hostname;
+        for (const firstParty in this.firstPartyWhiteList) {
+          const firstPartyFromList = new RegExp(firstParty);
```

Comment:
> Do you mean why can't I use something like whitelist.hasOwnProperty(firstParty)?

Yeah or `thirdParty.includes(thing)` for the other check.

> But blah.google.com is google.com, so shouldn't it be subject to the same whitelist ignore check? Why would we only be interested in www?

`www` is often treated as special for extensions like ours. I'm pretty confused why the list has `www.` and without if they are expecting different matching there. Like is it domain names or is it a hostname list?
I don't own the list so not sure, I would rather hope we could just duplicate `google.com` to `www.google.com` in the list and use an object key access and array includes rather than looping through all first parties if we can help it.

>  Would that solve it?

Technically no, it wouldn't account for left to right domains I don't think.
In order my preference:
- Key and array access
- string access modifying `www.google.com` to be `google.com` then comparing that inside the loop to array items
- constructing regexp at modification time
- improving what you have here

Either way if you use RegEx as you mentioned it will need to be escaped.